src/System.Runtime/tests/System/DecimalTests.cs

```diff
@@ -1293,5 +1295,375 @@ public static void DecrementOperator(decimal d, decimal expected)
         {
             Assert.Equal(expected, --d);
         }
+
+        [Fact]
+        public static void RandomBigInteger_Compare()
+        {
+            var values = GetRandomData(out var bigValues);
+            for (int i = 0, b = 0; i < values.Length; i++)
+            {
+                var d1 = values[i];
+                var b1 = bigValues[i];
+                for (int j = 0; j < values.Length; j++, b++)
+                {
+                    var d2 = values[j];
+                    var expected = b1.CompareTo(bigValues[j]);
+                    var actual = d1.CompareTo(d2);
+                    if (expected != actual)
+                        throw new Xunit.Sdk.AssertActualExpectedException(expected, actual, d1 + " CMP " + d2);
+                }
+            }
+        }
+
+        [Fact]
+        public static void RandomBigInteger_Add()
+        {
+            var overflowBudget = 1000;
+            var values = GetRandomData(out var bigValues);
+            for (int i = 0, b = 0; i < values.Length; i++)
+            {
+                var d1 = values[i];
+                var b1 = bigValues[i];
+                for (int j = 0; j < values.Length; j++, b++)
+                {
+                    var d2 = values[j];
+                    var expected = b1.Add(bigValues[j], out var expectedOF);
+                    if (expectedOF)
+                    {
+                        if (--overflowBudget < 0)
```

Comment:
What's this for?

Reply:
This is for limiting the number of overflow cases verified - without the limit there would be about 100k additional exceptions thrown (4-5s additional test execution time).